tube/spark/translator.py

```diff
@@ -161,8 +163,44 @@ def get_direct_children(self, root_df):
             root_df = root_df.leftOuterJoin(child_df).mapValues(lambda x: merge_and_fill_empty_fields(x, n.fields))
         return root_df
 
+    def get_types(self):
```

Comment:
I think we should move this part to the parser and save it as a property of parser.

Reply:
Will do.